### ISSUE TITLE
Multi-disk storage, data storage is unbalanced

### ISSUE THREAD

**Issue by xujinlong0314** [ClickHouse/ClickHouse#38372]
```
<cold>
                <disk>cold1</disk>
                <disk>cold2</disk>
                <disk>cold3</disk>
                <disk>cold4</disk>
                <disk>cold6</disk>
                <disk>cold7</disk>
</cold>
<move_factor>0.06</move_factor>
```

cold1 used 93%
cold2 used 40% 

data insert exception

not enough space cold1 ..G

how to balance or auto insert cold2 skip cold1

**Comment by alexey-milovidov** [comment]
I recommend creating RAID with Linux mdadm. It has better performance than JBOD due to striping.

**Comment by amosbird** [comment]
There is a merge tree setting `min_bytes_to_rebalance_partition_over_jbod` to rebalance data over JBOD volume.

**Comment by den-crane** [comment]
Addressed by 

ClickHouse release 22.6 https://github.com/ClickHouse/ClickHouse/blob/master/CHANGELOG.md
* Implement least_used load balancing algorithm for disks inside volume https://github.com/ClickHouse/ClickHouse/pull/36686

**Comment by alexey-milovidov** [comment]
@den-crane @amosbird Should we set the `least_used` by default?

**Comment by amosbird** [comment]
@alexey-milovidov I'm not sure. `least_used` will likely to have a balanced JBOD globally, however it can lead to significant data skew table-wise. All new data will end up in the newly added disk.

`min_bytes_to_rebalance_partition_over_jbod` is partition-wise rebalancing, which seems more helpful but only works for MergeTree-family tables.

**Comment by alexey-milovidov** [comment]
@amosbird What if we will place it randomly with probability proportional to the amount of empty space?

**Comment by amosbird** [comment]
I don't think that will help with the data skew problem. Data rebalancing usually happens with `merge` or `TTL`. I don't see how it's possible by only setup a disk selection strategy to solve the problem.

**Comment by xujinlong0314** [comment]
> There is a merge tree setting `min_bytes_to_rebalance_partition_over_jbod` to rebalance data over JBOD volume.
thanks,  min_bytes_to_rebalance_partition_over_jbod , Whether this applies only to new data writes,merge and TTL,  Historical data cannot be automatically balanced?

**Comment by xujinlong0314** [comment]
> Addressed by
> 
> ClickHouse release 22.6 https://github.com/ClickHouse/ClickHouse/blob/master/CHANGELOG.md
> 
> * Implement least_used load balancing algorithm for disks inside volume [Implement least_used load balancing algorithm for disks inside volume #36686](https://github.com/ClickHouse/ClickHouse/pull/36686)

thanks,  but i used lts version, how time merge to lts version

**Comment by alexey-milovidov** [comment]
@xujinlong0314 This setting only exists in the stable version 22.6.

**Comment by tomershafir** [comment]
@alexey-milovidov @amosbird I think non MergeTree engines are irrelevant: integration family doesnt store data, log family use cases should be satisfied by a single disk, and special ones dont attach volumes.

So maybe can default to `least_use` together with `min_bytes_to_rebalance_partition_over_jbod` calculated from jbod, with static assertion of the above assumption somehow.
